crates/rolldown/src/ast_scanner/side_effect_detector.rs

```diff
@@ -161,23 +164,29 @@ impl<'a> SideEffectDetector<'a> {
       | Expression::TSTypeAssertion(_)
       | Expression::TSNonNullExpression(_)
       | Expression::TSInstantiationExpression(_) => unreachable!("ts should be transpiled"),
-
+      Expression::BinaryExpression(binary_expr) => {
+        // For binary expressions, both sides could potentially have side effects
+        self.detect_side_effect_of_expr(&binary_expr.left)
+          || self.detect_side_effect_of_expr(&binary_expr.right)
+      }
+      Expression::PrivateInExpression(private_in_expr) => {
+        self.detect_side_effect_of_expr(&private_in_expr.right)
+      }
+      Expression::MetaProperty(_) | Expression::ThisExpression(_) => false,
+      Expression::AwaitExpression(await_expr) => {
+        self.detect_side_effect_of_expr(&await_expr.argument)
```

Comment:
Consider case like
```
let thenable = {
  then() {
     console.log('hello')
   }
}

await thenable
```

Reply:
ok

Reply:
The modification code for 'AwaitExpression' has been removed.

Reply:
> Consider case like
> 
> ```js
> let thenable = {
>   then() {
>      console.log('hello')
>    }
> }
> 
> await thenable
> ```

 The modification code for 'AwaitExpression' has been removed.

Reply:
Yeah. But it has side effect, so it should be true.